File: catalog/backends/java-quarkus/patterns/SlowQueryRepository.java
package com.example.service;
import jakarta.enterprise.context.ApplicationScoped;
import jakarta.inject.Inject;
import javax.sql.DataSource;
import java.sql.*;
import java.util.*;

// Pattern: DBM slow query — artificial delay via pg_sleep
// Adapt: replace 'your_table' with a domain entity table
@ApplicationScoped
public class SlowQueryRepository {
    @Inject DataSource ds;
    public List<Map<String, Object>> findAllSlow() throws SQLException {
        try (Connection c = ds.getConnection();
             ResultSet rs = c.createStatement().executeQuery(
                 "SELECT *, pg_sleep(0.3) FROM your_table ORDER BY created_at DESC LIMIT 50")) {
            List<Map<String, Object>> rows = new ArrayList<>();
            ResultSetMetaData md = rs.getMetaData();
            while (rs.next()) {
                Map<String, Object> row = new HashMap<>();
                for (int i = 1; i <= md.getColumnCount(); i++) row.put(md.getColumnName(i), rs.getObject(i));
                rows.add(row);
            }
            return rows;
        }
    }
}

File: catalog/backends/python-flask/patterns/slow_query_repository.py
import psycopg2
import os

# Pattern: DBM slow query — artificial delay via pg_sleep
# Adapt: replace 'your_table' with a domain entity table
DB_URL = os.environ.get("DATABASE_URL", "postgresql://demo:demo@postgresql:5432/demo")


def find_all_slow():
    conn = psycopg2.connect(DB_URL)
    cur = conn.cursor()
    cur.execute("SELECT *, pg_sleep(0.3) FROM your_table ORDER BY created_at DESC LIMIT 50")
    rows = cur.fetchall()
    cur.close()
    conn.close()
    return rows
